pkg/autodiscovery/common/utils/kubelet.go

```diff
@@ -16,3 +20,35 @@ func GetCustomCheckID(annotations map[string]string, containerName string) (stri
 	id, found := annotations[fmt.Sprintf(checkIDAnnotationFormat, containerName)]
 	return id, found
 }
+
+// ValidateAnnotationsMatching detects if AD annotations don't match a valid container identifier
+func ValidateAnnotationsMatching(annotations map[string]string, containerIdentifiers map[string]bool, containerNames map[string]bool, adPrefix string) []error {
+	var errors []error
+	adAnnotation := fmt.Sprintf(`%s.+\..+`, adPrefix)
+	checkIDAnnotation := fmt.Sprintf(checkIDAnnotationFormat, ".+\\")
```

Comment:
Are these supposed to match anywhere in the annotation? If not, regular expressions would need `^` and `$` around them to make sure they match the entire string.